google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb

```diff
@@ -167,13 +168,14 @@ def stop
         # @return [Subscriber] returns self so calls can be chained.
         #
         def wait! timeout = nil
-          wait_pool = synchronize do
-            @stream_pool.map do |stream|
-              Thread.new { stream.wait! timeout }
+          @final_stopped_thread ||= synchronize do
```

Comment:
* This is now identical to the code in `#stop`. Can we pull this into a helper method, to ensure we don't mess up in the future and alter one without altering the other?
* Looking at this, I think the synchronize block needs to surround the `@final_stopped_thread ||=` logic. Otherwise it's still possible for multiple threads to be started (and only one of them stored in `@final_stopped_thread`).

Reply:
Updated, PTAL.